PVSSpringBoot/src/test/java/com/example/PVSSpringBoot/PetTest/PetManagementServiceTest.java

```diff
@@ -0,0 +1,244 @@
+package com.example.PVSSpringBoot.PetTest;
+
+import org.assertj.core.api.Assertions;
+import static  org.assertj.core.api.Assertions.assertThat;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static  org.mockito.Mockito.verify;
+
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import com.example.PVSSpringBoot.Entities.Pet;
+import com.example.PVSSpringBoot.repositories.PetRepository;
+import com.example.PVSSpringBoot.services.PetManagementService;
+
+import java.sql.Date;
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@SpringBootTest
+@ExtendWith(MockitoExtension.class)
+public class PetManagementServiceTest {
+    @Mock
+    PetRepository repo;
+    @Autowired
+    @InjectMocks
+    PetManagementService service;
+
+    @Test
+    void testAddingAPet(){
+        Pet pet = Pet.builder()
+                .name("pet")
+                .type("cat")
+                .breed("persian")
+                .birthDate(Date.valueOf(LocalDate.EPOCH)).build();
+        Mockito.when(repo.existsInDB(pet.getName(), pet.getType(), pet.getBreed()))
+                .thenReturn(false);
+        Mockito.when(repo.save(pet)).thenReturn(pet);
+        assertThat(service.addPet(pet)).isTrue();
+        verify(repo,Mockito.times(1)).existsInDB(anyString(),anyString(),anyString());
+        verify(repo,Mockito.times(1)).save(pet);
+    }
+
+    @Test
+    void testAddingAnExistantPet(){
+        Pet pet = Pet.builder()
+                .name("pet")
+                .type("cat")
+                .breed("persian")
+                .birthDate(Date.valueOf(LocalDate.EPOCH)).build();
+        Mockito.when(repo.existsInDB(pet.getName(), pet.getType(), pet.getBreed()))
+                .thenReturn(true);
+        assertThat(service.addPet(pet)).isFalse();
+        verify(repo,Mockito.times(1)).existsInDB(anyString(),anyString(),anyString());
+    }
+
+    @Test
+    void testRemovePet(){
+        Mockito.when(repo.existsById(1L)).thenReturn(true);
+        assertThat(service.removePet(1L)).isTrue();
+        verify(repo,Mockito.times(1)).existsById(1L);
+    }
+    @Test
+    void testRemoveNonExistantPet(){
+        Mockito.when(repo.existsById(1L)).thenReturn(false);
+        assertThat(service.removePet(1L)).isFalse();
+        verify(repo,Mockito.times(1)).existsById(1L);
+
+    }
+
+    @Test
+    void testGetById(){
```

Comment:
Is this test fail ? As this pet is removed in the previous test.

Reply:
No DB is used as the behaviour of the repository is mocked and there is no state